src/registry/radio/helpers.ts

```diff
@@ -0,0 +1,11 @@
+import {RadioComponentSchema} from '@open-formulieren/types';
+import {Option} from '@open-formulieren/types/lib/formio/common';
+
+// A type guard is needed because TS cannot figure out it's a discriminated union
+// when the discriminator is nested.
+// See https://github.com/microsoft/TypeScript/issues/18758
+export const checkIsManualOptions = (
+  component: RadioComponentSchema
+): component is RadioComponentSchema & {values: Option[] | undefined} => {
+  return component.openForms.dataSrc === 'manual';
+};
```

Comment:
Can't we have this generic? Something along the lines of

```suggestion
export const checkIsManualOptions<T> = (
  component: T
): component is T & {values: Option[] | undefined} => {
  return component.openForms.dataSrc === 'manual';
};
```

Reply:
no because `select` used a different shape for the `values`, it's nested inside `data`. So I consider moving this further away from where it's used and adding generics too much indirection and a little bit of duplicated code like this is totally acceptable to me, but if you have strong feelings about it I'm willing to reconsider

Reply:
Ok now makes sense if structure is different, better to have duplicated code rather than hard to understand generics :)